Homework_Josh_Smith

joshuaPSmith opened the pull request.

Commit:
Homework

Commit:
Added eslint and gitignore

jamdon commented:
There's an issue with your `/food-types/:type` route, but I'm unable to leave inline comments because there are too many files in your repo. This is because node_modules was included. I see you added a .gitignore, but it must have been added after pushing up the node_modules to your repo. What you want to do is completely delete them (the entire folder and everything in it), then push that change up to your repo. Once you've done that, I can leave an inline comment about the problem route.

joshuaPSmith commented:
I will remove it and repush. Can you explain what the gitignore actually
does?

Also can you share the modification you made on your lint file for the
server code?

Thanks!

On Fri, May 13, 2016 at 3:52 AM, jamdon notifications@github.com wrote:

> There's an issue with your /food-types/:type route, but I'm unable to
> leave inline comments because there are too many files in your repo. This
> is because node_modules was included. I see you added a .gitignore, but it
> must have been added after pushing up the node_modules to your repo. What
> you want to do is completely delete them (the entire folder and everything
> in it), then push that change up to your repo. Once you've done that, I can
> leave an inline comment about the problem route.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/JSCRIPT300-Spring2016/Module_4/pull/15#issuecomment-218987406

## 

Josh Smith
joshuap.smith12@gmail.com
(205) 960-5611

Commit:
removed node_modules

jamdon commented:
The .gitignore file tells git to not track anything listed inside the file. If you add `node_modules` to the file, it means git will ignore the entire node_modules folder, and it won't be part of your commit and subsequent push to your remote repository. As for linting server code, I'm not sure what you mean.

jamdon commented on trucks.js:
```diff
@@ -1,3 +1,4 @@
+var _ = require("underscore");
```
You're referencing underscore here, but your package.json file lists lodash as a dependency. Change this from underscore to lodash here (currently server won't run because underscore isn't found.)

jamdon commented:
Ok, the node_modules are gone, so I was able to leave a comment inline on what's preventing your server from running (referencing underscore instead of lodash)

Commit:
added lodash instead of underscore

jamdon commented on index.js:
```diff
@@ -1,4 +1,34 @@
+
+app.get('food-types/:type', function (request, response) {
+  var type = request.params.type;
+  var truckList = trucks.filterTrucksByFoodType(typr);
```
There's a typo here preventing your route from working correctly.

jamdon commented on index.js:
```diff
@@ -1,4 +1,34 @@
+  var truckList = trucks.getTrucks();
+
+  response.send(truckList);
+})
```
don't forget your `;`

jamdon commented on index.js:
```diff
@@ -1,4 +1,34 @@
+  var truck = trucks.getTruck(request.params.name);
+
+  response.send(truck);
+})
```
don't forget your `;`

jamdon commented on index.js:
```diff
@@ -1,4 +1,34 @@
+  var foodList = trucks.getFoodTypes();
+
+  response.send(foodList);
+})
```
don't forget your `;`

jamdon commented on index.js:
```diff
@@ -1,4 +1,34 @@
+  var truckList = trucks.filterTrucksByFoodType(typr);
+
+  response.send(truckList);
+})
```
don't forget your `;`

jamdon commented on routes/trucksRoutes.js:
```diff
@@ -0,0 +1,12 @@
+  .get(function (request, response) {
+    var trucks = trucksList.getTrucks();
+    response.status(200).json(trucks);
+  })
```
don't forget your `;`

jamdon commented:
I've left another inline comment regarding a small issue in one of your routes.

Commit:
Fixed typos

Still can't get the food-types/:type route to work

jamdon commented on routes/trucksRoutes.js:
```diff
@@ -0,0 +1,12 @@
+
+var router = express.Router();
+
+router.route('/')
```
btw, the router isn't necessary for this assignment

jamdon commented on trucks.js:
```diff
@@ -503,9 +504,48 @@ var foodTrucks = [
+      return type.toLowerCase();
+    });
+
+    return truck.type.indexOf(foodType.toLowerCase) !== -1;
```
The problem here is that you converted the foodType to lowercase, but you're comparing against a list that hasn't been converted to lowercase. You're not really comparing to the `types` list that you just lowercased. Instead of calling indexOf on truck.type, call it on the types array.

jamdon commented:
Very close. I left one inline comment.

Commit:
Changed the type compared based on inline commit

It is still not working correctly but I will rewatch the video and get
it figured out

jamdon commented on index.js:
```diff
@@ -1,4 +1,34 @@
+  response.send(foodList);
+});
+
+app.get('food-types/:type', function (request, response) {
```
Your route is missing the leading '/', but works great when I put it in there.

jamdon commented:
There was one very small issue causing the /food-types/:type route to not get hit (see inline comment) - don't forget to test your work! 👍
